src/viking.js

```diff
@@ -1,14 +1,107 @@
 // Soldier
-class Soldier {}
+class Soldier {
+
+  constructor(health, strength) {
+    this.health = health;
+    this.strength = strength;
+  }
+
+  attack() {
+    return this.strength;
+  }
+
+  receiveDamage(damage) {
+    this.health -= damage;
+  }
+
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+
+  constructor(name, health, strength) {
+    super(health, strength);
+    this.name = name;
+  }
+
+  receiveDamage(damage) {
+    this.health -= damage;
```

Comment:
if you see in soldier class we have the same method right? instead of writing line 28 we can use
this.receiveDamage(damage)

Not its only one line but you can imagine what if its bigger"normaly is bigger"